### ISSUE TITLE
Support using nvidia-docker with --docker options

### ISSUE THREAD

**Issue by fkm3** [commercialhaskell/stack#3499]
Would it be reasonable to add an option to control the path used to invoke docker so that nvidia-docker can be easily used instead? See https://github.com/tensorflow/haskell/pull/157 for motivating example.

Thanks!

**Comment by mgsloan** [comment]
Makes sense, this falls under the umbrella of https://github.com/commercialhaskell/stack/issues/2369
